File: 2840.check-if-strings-can-be-made-equal-with-operations-ii/solution.kt
class Solution {
    fun checkStrings(s1: String, s2: String): Boolean {
        return checkFrom(s1, s2, 0) && checkFrom(s1, s2, 1)
    }

    private fun checkFrom(s1: String, s2: String, start: Int): Boolean {
        val h = IntArray(26)
        for (i in start..s1.lastIndex step 2) {
            h[s1[i] - 'a']++
            h[s2[i] - 'a']--
        }
        return h.all { it == 0 }
    }
}

File: 2840.check-if-strings-can-be-made-equal-with-operations-ii/solution.go
func checkStrings(s1 string, s2 string) bool {
    return check(s1, s2, 0) && check(s1, s2, 1)
}

func check(s1 string, s2 string, start int) bool {
    var h [26]int

    for i := start; i < len(s1); i += 2 {
        h[s1[i] - 'a']++
        h[s2[i] - 'a']--
    }
    for _, c := range h {
        if c != 0 {
            return false
        }
    }
    return true
}